src/cdk-experimental/menu/menu-item-trigger.ts

```diff
@@ -72,34 +88,96 @@ export class CdkMenuItemTrigger implements OnDestroy {
   /** Open/close the attached menu if the trigger has been configured with one */
   toggle() {
     if (this.hasMenu()) {
-      this.isMenuOpen() ? this._closeMenu() : this._openMenu();
+      this.isMenuOpen() ? this.closeMenu() : this.openMenu();
+    }
+  }
+
+  /** Open the attached menu. */
+  openMenu() {
+    if (!this.isMenuOpen()) {
+      this.opened.next();
+
+      this._overlayRef = this._overlay.create(this._getOverlayConfig());
+      this._overlayRef.attach(this._getPortal());
+    }
+  }
+
+  /** Close the opened menu. */
+  closeMenu() {
+    if (this.isMenuOpen()) {
+      this.closed.next();
+
+      this._overlayRef!.detach();
     }
   }
 
   /** Return true if the trigger has an attached menu */
   hasMenu() {
-    return !!this._menuPanel;
+    return !!this.menuPanel;
   }
 
   /** Whether the menu this button is a trigger for is open */
   isMenuOpen() {
     return this._overlayRef ? this._overlayRef.hasAttached() : false;
   }
 
-  /** Open the attached menu */
-  private _openMenu() {
-    this.opened.next();
-
-    this._overlayRef = this._overlay.create(this._getOverlayConfig());
-    this._overlayRef.attach(this._getPortal());
+  /**
+   * Get a reference to the rendered Menu if the Menu is open and it is visible in the DOM.
+   * @return the menu if it is open, otherwise undefined.
+   */
+  getMenu(): Menu | undefined {
+    return this.menuPanel?._menu;
   }
 
-  /** Close the opened menu */
-  private _closeMenu() {
-    if (this.isMenuOpen()) {
-      this.closed.next();
-
-      this._overlayRef!.detach();
+  /**
+   * Handles keyboard events for the menu item, specifically opening/closing the attached menu and
+   * focusing the appropriate submenu item.
+   * @param event the keyboard event to handle
+   */
+  _toggleOnKeydown(event: KeyboardEvent) {
+    const keyCode = event.keyCode;
+    switch (keyCode) {
+      case SPACE:
+      case ENTER:
+        event.preventDefault();
+        this.toggle();
+        this.menuPanel?._menu?.focusFirstItem('keyboard');
+        break;
+
+      case RIGHT_ARROW:
+        if (this._isParentVertical()) {
+          event.preventDefault();
+          if (this._directionality.value === 'ltr') {
```

Comment:
The `_directionality` should be decorated with `@Optional`, because we aren't guaranteed for it to exist. Also generally we format direction checks like this, because LTR is the most common case and it handles undefined values better:

```
if (direction === 'rtl') {
  // do RTL stuff
} else {
  // do LTR stuff
}
```